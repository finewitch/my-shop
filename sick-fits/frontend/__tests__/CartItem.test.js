import CartItemComponent from '../components/CartItem';
import {shallow, mount} from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
id: "ck4hjcero355k0993dhxylnv9",
item: {price: 1213, title: "fotka", description: "123123"},
quantity: 1
}

describe('<CartItemComponent/>',()=>{

    it('renders', ()=>{
        shallow(<CartItemComponent item={fakeItem}/>)
    })

    it('matches the snapshop', ()=>{
        const wrapper = shallow(<CartItemComponent item={fakeItem}/>)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('updates via props', ()=>{

        const wrapper = shallow(<CartItemComponent item={fakeItem}/>)
        expect(toJSON(wrapper)).toMatchSnapshot()
        wrapper.setProps({item:{...fakeItem, quantity: 2} })
        expect(toJSON(wrapper)).toMatchSnapshot()

    })
    
})