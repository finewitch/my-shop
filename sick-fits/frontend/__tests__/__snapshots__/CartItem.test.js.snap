// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItemComponent/> matches the snapshop 1`] = `
<div
  className="cart-product"
>
  <div
    className="cart-product-title"
  >
    $12.13
  </div>
  <div
    className="cart-product-index"
  >
    1
     x 
    $12.13
  </div>
  <div
    className="cart-product-price"
  >
    fotka
  </div>
  <RemoveFromCart
    id="ck4hjcero355k0993dhxylnv9"
  />
</div>
`;

exports[`<CartItemComponent/> updates via props 1`] = `
<div
  className="cart-product"
>
  <div
    className="cart-product-title"
  >
    $12.13
  </div>
  <div
    className="cart-product-index"
  >
    1
     x 
    $12.13
  </div>
  <div
    className="cart-product-price"
  >
    fotka
  </div>
  <RemoveFromCart
    id="ck4hjcero355k0993dhxylnv9"
  />
</div>
`;

exports[`<CartItemComponent/> updates via props 2`] = `
<div
  className="cart-product"
>
  <div
    className="cart-product-title"
  >
    $24.26
  </div>
  <div
    className="cart-product-index"
  >
    2
     x 
    $12.13
  </div>
  <div
    className="cart-product-price"
  >
    fotka
  </div>
  <RemoveFromCart
    id="ck4hjcero355k0993dhxylnv9"
  />
</div>
`;
